### PR TITLE
Move tests to rust

### PR DESCRIPTION
**Pull request by guibescos**
This PR aims to move the last python tests to rust.
I had to refactor parts of pyth_simulator.rs to support moving the clock and to support sending multiple instruction in one transaction.
The tests imitate what was happening in Python but with some extra assertions.
I will delete the python tests once this gets merged.

### CODE REVIEW

File: program/rust/src/processor/upd_price.rs [hunk 1]
```diff
@@ -145,18 +142,11 @@ pub fn upd_price(
         }
     }
 
+
     // Try to update the publisher's price
     if is_component_update(cmd_args)? {
-        let mut status: u32 = cmd_args.status;
-        let mut threshold_conf = cmd_args.price / MAX_CI_DIVISOR;
```
Comment (guibescos) [inline program/rust/src/processor/upd_price.rs:151]: This code gets refactored so I can use it in tests as well.

File: program/rust/src/tests/pyth_simulator.rs [hunk 2]
```diff
@@ -64,7 +76,7 @@ use {
 /// this struct to test how pyth instructions execute in the Solana runtime.
 pub struct PythSimulator {
     program_id:            Pubkey,
-    banks_client:          BanksClient,
+    context:               ProgramTestContext,
```
Comment (guibescos) [inline program/rust/src/tests/pyth_simulator.rs:79]: I need this refactor to be able to advance the clock in the simulator

File: program/rust/src/tests/pyth_simulator.rs [hunk 3]
```diff
@@ -149,17 +177,18 @@ impl PythSimulator {
     }
 
 
-    /// Process a transaction containing `instruction` signed by `signers`.
+    /// Process a transaction containing `instructions` signed by `signers`.
     /// `payer` is used to pay for and sign the transaction.
-    async fn process_ix(
+    async fn process_ixs(
```
Comment (guibescos) [inline program/rust/src/tests/pyth_simulator.rs:182]: This function got updated to support multiple instructions in the same transaction

File: program/rust/src/tests/test_publish.rs [hunk 4]
```diff
@@ -0,0 +1,164 @@
+use {
```
Comment (guibescos) [inline program/rust/src/tests/test_publish.rs:1]: Will replace test_update_price.py

File: program/rust/src/tests/test_publish_batch.rs [hunk 5]
```diff
@@ -0,0 +1,128 @@
+use {
+    crate::{
+        accounts::PriceAccount,
```
Comment (guibescos) [inline program/rust/src/tests/test_publish_batch.rs:3]: Will replace test_update_price.py

File: program/rust/src/tests/test_publish_batch.rs [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+            check_confidence_too_big(quote.price, quote.confidence, quote.status).unwrap()
+        );
+        assert_eq!(price_data.comp_[0].agg_.price_, 0);
+        assert_eq!(price_data.comp_[0].agg_.conf_, 0);
+        assert_eq!(price_data.comp_[0].agg_.status_, PC_STATUS_UNKNOWN);
+    }
+
+    sim.warp_to_slot(2).await.unwrap();
+
+    let mut new_quotes: HashMap<String, Quote> = HashMap::new();
```
Comment (guibescos) [inline program/rust/src/tests/test_publish_batch.rs:91]: In `test_update_price.py`, the second set of quotes are (1,1,"TRADING"). I changed the second set of quotes to be random here.

File: program/rust/src/utils.rs [hunk 7]
```diff
@@ -195,6 +199,25 @@ pub fn is_component_update(cmd_args: &UpdPriceArgs) -> Result<bool, OracleError>
     }
 }
 
+// Return PC_STATUS_IGNORED if confidence is bigger than price divided by MAX_CI_DIVISOR else returns status
+pub fn check_confidence_too_big(
```
Comment (jayantk) [inline program/rust/src/utils.rs:203]: this function name sounds like it's going to error if the confidence is too big. something more like `get_status_for_update` would be better. (I don't love that name either, but the point being the name should indicate that this returns a status)

**Review by jayantk — APPROVED** [review]
cool. i left you a very minor comment. feel free to merge this and rename that function in whatever your next pr is.

**Pull request merged**